### DISCUSSION TITLE
Simple command in Erlang shell not working

### DISCUSSION THREAD

**Discussion by functonal** [gleam-lang/gleam#1479] [Questions & support] [answered]
Followed the web link for creating a simple project in Gleam  https://gleam.run/writing-gleam/creating-a-project/  
which basically translates to : 

```
gleam new my_fantastic_library
cd my_fantastic_library
gleam test my_fantastic_library
gleam run
```
the output was:
```
Compiling my_fantastic_library
Compiled in 0.81s
Running my_fantastic_library.main
Hello from my_fantastic_library!
```
All fine till now. 👍
I invoked the  Erlang shell from here as per https://gleam.run//writing-gleam/using-the-project/
```
➜  my_fantastic_library gleam shell
  Compiling my_fantastic_library
   Compiled in 1.70s
    Running Erlang shell
Erlang/OTP 24 [erts-12.2.1] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [jit] [dtrace]

Eshell V12.2.1  (abort with ^G)
1>my_fantastic_library:hello_world().
** exception error: undefined function my_fantastic_library:hello_world/0
2>
```
I tried several times on macOS,Ubuntu and Windows 10 : The result is same
Why it is showing error?
What are the things I have to look to troubleshoot this?

Thanks

**Comment by lpil** [comment, marked as answer]
Hi @functonal ! The generated function in new projects is called `main`, so it'll be `my_fantastic_library:main().` .

Looks like those docs are out of date, I'll fix that now.

**Reply by functonal** [reply]
Thanks a lot, that  famous main() 
1> my_fantastic_library:main().
Hello from my_fantastic_library!

This was a single file .gleam project: so main() ,
What about when multiple files will be in project  and I will need to show result from any particular function : How would I invoke then ,could you please guide..

Thanks again.

**Reply by lpil** [reply]
The Erlang syntax for a function call is `modulename:functionname().`, so if you have a module called `one` with a function called `two` it would be `one:two().`.

One last thing to keep in mind is that the `/` in Gleam module names is replaced with a `@` in Erlang, so the module `one/two` becomes `one@two`.

If you'd like to use the shell a lot it'd be worth learning a little Erlang as it is not a Gleam shell currently. If you'd rather work in Gleam you could call your function from your main function or call it from your tests.

**Reply by functonal** [reply]
Thanks,I was expecting this..As of my Erlang and Elixir I have forwarded quite a bit.
Just like Elixir and LFE (also have some knowledge) as I am inclined to Lisp/Scheme other than the mainstream languages.,I wanted to know the expression clarity for module:functionname().I found out in your answer my actual query : that gleam still follows the Erlang strictly, just like  Elixir(gone thru the source code of Elixir in Erlang) and LFE.

The answer is clear  to me.

Thanks again.
